### ISSUE TITLE
consider selecting for the single test case that covers the most code

### ISSUE THREAD

**Issue by shapr** [shapr/trynocular#8]
After reading https://www.cs.tufts.edu/comp/150FP/archive/john-hughes/beginners-luck.pdf and https://lcamtuf.blogspot.com/2014/11/pulling-jpegs-out-of-thin-air.html I think we do want the ability to cover the most code with a single input.
I suspect that will be a good way to find edge cases and surprising inputs.

**Comment by cdsmith** [comment]
Yeah, my first thought is there's a good case for allowing some configurability in what the objective function is.

My second thought is that even in the traditional case, where you're trying to maximize coverage across all test cases instead of just a single test case, there's still some value in getting extra coverage for code that was already covered. That coverage is just worth less the more times it has already run.  So what we really want is something like

```
tixRegionValue :: Int -> Int -> Double
tixRegionValue before after = ???

tixModuleValue :: TixModule -> TixModule -> Double
tixModuleValue (TixModule _ _ _ before) (TixModule _ _ _ after) = sum (zipWith tixRegionValue before after)

tixValue :: Tix -> Tix -> Double
tixValue (Tix before) (Tix after) = sum (zipWith tixModuleValue before after)
```

So what should `tixRegionValue` be?  Our current code is, essentially, `if before == 0 and after > 0 then 1 else 0`, also known as `min 1 after - min 1 before`.  But another interesting choice might be `sqrt after - sqrt before`, or `log1p after - log1p before`, or `(\x -> 1 - 1 / (1 + x)) after - (\x -> 1 - 1 / (1 + p)) before`, or any other function there with a positive first derivative (i.e., more coverage is better) but a negative second derivative (i.e., additional coverage becomes less valuable if there's been plenty of coverage already).

In this case, your proposal is to set `tixRegionValue before after = min 1 (after - before)`, so the second derivative is zero and coverage is never discounted.  That's also fine.
